### PR TITLE
Create image for use in ML pipelines

### PR DESCRIPTION
**Pull request by adamjstewart**
The idea is to create an image that can be used across all Linux ML pipelines (x86_64, aarch64, ppc64le) with a newer OS/GCC version. Some criteria:

1. TF requires GCC 9.4+
1. TF requires `/usr/bin/python3`
2. TF requires ROCm to be installed in a specific system directory

P.S. I don't know how to write or test a Dockerfile so if you have any suggestions, it's probably quicker to push to my branch than it is to explain it to me 😅

### CODE REVIEW

**Comment by adamjstewart** [comment]
@alalazo https://github.com/spack/spack/pull/43751 passed successfully so I think this image is good to go. There are still a few issues with https://github.com/spack/spack/pull/39666 but those are due to aarch64, not due to the image.

**Comment by kwryankrattiger** [comment]
Not sure if there is something more robust, but usually I just build the image locally, then run the target stack install on it to make sure it passes, and call it good.

I there are any changes to the spack.yaml that are needed, like pointing to the ROCm/Python externals, then you can make sure everything is working before pushing the image and it makes the Spack CI PR much easier.

**Comment by adamjstewart** [comment]
For now, the image I created here is good enough. It doesn't yet support external ROCm, but I'm told that AMD is working on supporting Spack-installed ROCm in the future. I'll leave ROCm to someone else.

File: Dockerfiles/ubuntu-22.04.dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  mercurial \
+  ncurses-dev \
+  patch \
+  patchelf \
+  pciutils \
+  python3-pip \
+  rsync \
+  unzip \
+  wget \
+  zlib1g-dev \
```
Comment (kwryankrattiger) [inline Dockerfiles/ubuntu-22.04.dockerfile:40]: Just to double check all of these are needed from the system.

```
autoconf
automake
gettext
libffi-deva
libssl-dev
libxml2-dev
m4
ncurses-dev
zlib1g-dev
```

**Review by kwryankrattiger — APPROVED** [review]
LGTM

**Review by alalazo — APPROVED** [review]

**Pull request merged**